### DISCUSSION TITLE
How to embed image from a url on my own server?

### DISCUSSION THREAD

**Discussion by lanyusan** [toeverything/blocksuite#5400] [Q&A]
I have gone through the docs and issues and found no clue.

My use case is that I have already got a pretty big stock image repository that is hosted on my own server. I wish to be able to insert  image to the document by its url. The image should be displayed as image in block flow, not as a link or card. 

How do I do it?

Thanks.

**Comment by doodlewind** [comment]
Thanks for your trying! Currently the `affine:image` block supports `sourceId` for image source by default, which is designed to match the requirement to be used with S3-like blob storage. But we have plan to support `src` field for URL-based image loading (which is pretty much something can be marked as `help wanted`), hope you can stay tuned 🙏
